### ISSUE TITLE
hlint no respecting "ignore Avoid restricted flags" in test/Spec.hs

### ISSUE THREAD

**Issue by JasonMFry** [ndmitchell/hlint#1321]
I suspect this is user error on my part, but I have tried everything I can think of and I'm still getting an hlint error when I don't want to. How can I change my hlint.yaml to ignore Avoid restricted flags in test/Spec.hs?

output of `hlint .`:
```text
test/Spec.hs|1| 1-68: Warning: Avoid restricted flags
|| Found:
||   {-# OPTIONS_GHC -F -pgmF hspec-discover -optF --module-name=Spec #-}
|| Note: may break the code, may break the code, may break the code, may break the code, may break the code
|| 
|| 1 hint
|| make: *** [hlint] Error 1
```

```haskell
-- test/Spec.hs
{-# OPTIONS_GHC -F -pgmF hspec-discover -optF --module-name=Spec #-}
```

```yaml
# .hlint.yaml
- ignore:
    name:
      - Avoid restricted flags
      # other things here, but probably irrelevant to this issue
    within:
      - "**.*Spec" # this works for all files named FooSpec.hs, but not Spec.hs
      - Spec # I have tried many things here, including Spec.hs, test/Spec.hs, test.Spec.hs, etc etc
```

**Comment by JasonMFry** [comment]
Closing in favor of https://github.com/ndmitchell/hlint/issues/1320. I have _no_ idea how I made two issues lol 🤦🏼
